apps/src/applab/dataBrowser/DataOverview.jsx

```diff
@@ -125,9 +81,28 @@ const DataOverview = React.createClass({
 });
 
 export default connect(state => ({
-  view: state.data.view
+  view: state.data.view,
+  tableListMap: state.data.tableListMap || {}
 }), dispatch => ({
   onViewChange(view, tableName) {
     dispatch(changeView(view, tableName));
+  },
+  onTableAdd(tableName) {
+    // Add the table name to the model even though it doesn't exist in the database yet,
+    // so that it appears in the overview even if we don't create a record in it.
+    // addTableName() is idempotent, so we won't double-count the table name later
+    // when the the first record is added to it.
+    dispatch(addTableName(tableName));
+
+    dispatch(changeView(DataView.TABLE, tableName));
+  },
+  onTableDelete(tableName) {
+    // Explicitly remove the table from the model to make sure it disappears from the
+    // list, since we won't receive a delete event from the database if the table did
+    // not exist in the database. This could happen when deleting a table after adding it
+    // without adding any records to it.
+    dispatch(deleteTableName(tableName));
+
+    FirebaseStorage.deleteTable(tableName);
```

Comment:
How you have it seems okay.

Actions are payloads of information. I wasn't suggesting put it in the action, but in the action creator, which is a little different. In some respects this feels analogous to an async web request, where we want to go perform some async action, and potentially update our store based on the result. Conceivably, we might want to display some UI if the call to FirebaseStorage.deleteTable resulted in some sort of async error. However, given that we're not doing anything with the result from FirebaseStorage.deleteTable for now, I think you're approach is sufficient and probably easier to grok.